nms/v1_19R1/src/main/java/net/dzikoysk/funnyguilds/nms/v1_19R1/message/V1_19R1MessageAccessor.java

```diff
@@ -40,11 +40,9 @@ public void sendTitleMessage(TitleMessage titleMessage, Collection<? extends Pla
 
     @Override
     public void sendActionBarMessage(String text, Player... players) {
-        PacketPlayOutChat actionBarPacket =
-                new PacketPlayOutChat(CraftChatMessage.fromStringOrNull(text, true), ChatMessageType.c, SENDER_ALWAYS_DISPLAY);
-
-        for (Player player : players) {
-            ((CraftPlayer) player).getHandle().b.a(actionBarPacket); // a -> sendPacket
+        //każde wywołanie NMS to jeden martwy kotek :(
+        for (Player p : players) {
+            p.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent(text));
```

Comment:
Moim zdaniem tu zdecydowanie powinno powinno zostać użycie pakietu - w końcu jak już możemy użyć NMS, to użyjmy, a nie robimy na około

Reply:
![obraz](https://user-images.githubusercontent.com/25874220/173563273-4c39b0c6-e205-41d0-8ff9-c229929cd1b7.png)
Od tego jest API żeby go używać a nie kopać się z NMS

Reply:
Bo wtedy jeszcze nie było takiego wyśmienitego API ;) Tamte metody i tak pod maską wysyłają pakiety więc nie rozumiem w czym rzecz, PacketPlayOutChat został zresztą usunięty, w 1.19 wchodzi szyfrowanie i kilka innych rzeczy więc te zmiany muszą zostać wprowadzone w spigocie który wyprowadza elegancko API i jeszcze musi je wprowadzić twórca każdego pluginu który używa NMS, boilerplating.